### PR TITLE
Adding base for sphinx-based documentation

### PR DESCRIPTION
**Pull request by andrsd**
This PR does:
- adds the foundation for generating doco using `sphinx`
- adds "Getting started pages"
- adds "Basic Tutorials"

### CODE REVIEW

**Comment by andrsd** [comment]
Brief instructions on how to use this (for testing, for example):

Install required python packages:

```
$ cd /path/to/opensn
$ pip install -r doc/requirements.txt
```

Build doco
```
$ cd <build-dir>
$ cmake ..
$ make doc
```

Note: you must have `doxygen` installed.

**Review by ragusa — APPROVED** [review]

File: doc/images/XSs/.$XSLogicDiagram.drawio.bkp
[the remark this answers is not in the document]
Comment (andrsd) [reply doc/images/XSs/.$XSLogicDiagram.drawio.bkp:1]: Done

**Review by zhardy-lanl — APPROVED** [review]
A definitely dumb question, but how do you toggle the `OPENSN_WITH_<name>` options when configuring? Is it like setting a `cmake` option like `-DOPENSN_WITH_<name>`?

**Comment by andrsd** [comment]
Use `-DCMAKE_OPTION=<value>` from command line. But, use `ccmake` (console front-end for configuring) if you forgot the name.

You will see something like this:
```
CMAKE_BUILD_TYPE                 Release                                                                                                                                                                        
CMAKE_COLOR_DIAGNOSTICS          ON                                                                                                                                                                             
CMAKE_INSTALL_PREFIX             /usr/local                                                                                                                                                                     
CMAKE_OSX_ARCHITECTURES                                                                                                                                                                                         
CMAKE_OSX_DEPLOYMENT_TARGET                                                                                                                                                                                     
CMAKE_OSX_SYSROOT                /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk                                                                                                                            
OPENSN_WITH_DOCS                 ON                                                                                                                                                                             
OPENSN_WITH_LUA                  ON                                                                                                                                                                             
PETSC_INCLUDE_DIR                /some/path/include                                                                                                                                   
PETSC_LIBRARY                    /some/path/lib/libpetsc.dylib                                                                                                                        
VTK_DIR                          /some/path/lib/cmake/vtk-9.3                                                                                                                         
VTK_MPI_NUMPROCS                 2                                                                                                                                                                              
```

Just navigate to the option and change it to what you want it to be.

**Pull request merged**